src/main.py

```diff
@@ -91,3 +125,80 @@ def send_error(self, category: str, error_msg: str, **kwargs):
     def send_stack_trace(self, category: str, stack_trace: str, **kwargs):
         if self.consent:
             self.sender.send(self.backend, self.backend.build_stack_trace_message(category, stack_trace, **kwargs))
+
+    @staticmethod
+    def _update_opt_in_status(tid: str, new_opt_in_status: bool):
+        """
+        Updates opt-in status.
+
+        :param tid: ID of telemetry base.
+        :param new_opt_in_status: new opt-in status.
+        :return: None
+        """
+        app_name = 'opt_in_out'
+        app_version = '1.0.0'
+        opt_in_checker = OptInChecker()
+        opt_in_check = opt_in_checker.check()
+
+        prev_status = OptInStatus.UNDEFINED
+        if opt_in_check == ISIPCheckResult.DECLINED:
+            prev_status = OptInStatus.DECLINED
+        elif opt_in_check == ISIPCheckResult.ACCEPTED:
+            prev_status = OptInStatus.ACCEPTED
+
+        if new_opt_in_status:
+            updated = opt_in_checker.update_result(ISIPCheckResult.ACCEPTED)
+        else:
+            updated = opt_in_checker.update_result(ISIPCheckResult.DECLINED)
+        if not updated:
+            return
+
+        telemetry = Telemetry(tid=tid, app_name=app_name, app_version=app_version)
+
+        if new_opt_in_status:
+            telemetry.backend.generate_new_uid_file()
+            if opt_in_check != ISIPCheckResult.ACCEPTED:
+                telemetry.send_opt_in_event(OptInStatus.ACCEPTED, prev_status)
+            print("You have successfully opted in to send the telemetry data.")
+        else:
+            if opt_in_check != ISIPCheckResult.DECLINED:
```

Comment:
it's better to use prev_status here, it's more readable

Reply:
Done.